test/test-network/systemd-networkd-tests.py

```diff
@@ -5031,37 +5031,56 @@ def test_keep_configuration_on_restart(self):
         # Add an unmanaged interface with an up address
         call('ip link add unmanaged0 type dummy')
         call('ip link set unmanaged0 up')
+        call('ip -4 addr add 10.20.30.40/32 dev unmanaged0')
         call('ip -6 addr add 2001:db8:9999:f101::15/64 dev unmanaged0')
 
-        copy_network_unit('12-dummy.netdev', '85-static-ipv6.network')
-        start_networkd()
-        self.check_keep_configuration_on_restart()
-
         # Start `ip monitor` with output to a temporary file
-        with tempfile.TemporaryFile(mode='r+', prefix='ip_monitor') as logfile:
-            process = subprocess.Popen(['ip', 'monitor', 'dev', 'dummy98'], stdout=logfile, text=True)
-            restart_networkd()
+        with tempfile.TemporaryFile(mode='r+', prefix='ip_monitor_u') as logfile_unmanaged:
+            process_u = subprocess.Popen(['ip', 'monitor', 'dev', 'unmanaged0'], stdout=logfile_unmanaged, text=True)
+
+            copy_network_unit('12-dummy.netdev', '85-static-ipv6.network', '85-unmanaged.link')
```

Comment:
The .link file should be copied before creating `unmanaged0` interface, as it is used by udevd, rather than networkd.

I think it is OK to move whole copy_network_unit() line at the beginning of the test function, as copying .netdev and .network does not affect anything unless you restart or reload networkd.